### ISSUE TITLE
Opening a file results in re-attaching LSP clients even if they are already attached

### ISSUE THREAD

**Issue by GregoRCodi** [kyazdani42/nvim-tree.lua#517]
Hey,

I have noticed the following issue: when opening a file through `nvim-tree` using `o` or `<C-t>`, the lsp clients are re-attached even if the file is already open. The problem does not happen when using `<C-x>` or `<C-v>`.`

In the following video, you can see that the message `tsserver attached` is shown whenever I go to nvim-tree and press `o` to open the file again (the file is already open as the only file in the main window). There are no more messages when I'm reopening the file using Telescope

https://user-images.githubusercontent.com/84332110/126040554-bef40697-3e81-471c-9fad-2f9dca5df805.mp4

Here's how it works using `<C-x>` and `<C-v>` (no unnecessary re-attaches):

https://user-images.githubusercontent.com/84332110/126040621-236ae927-87ce-44cb-ab55-27ffec8be257.mp4

I have decided to report an issue for nvim-tree, because it only happens for some keybindings, and because it does not happen when opening files using Telescope.

**Comment by gegoune** [comment]
I wonder whether this has anything to do with my blinking cursor issue, you can see it on first video here too.
https://github.com/kyazdani42/nvim-tree.lua/issues/451

**Comment by kyazdani42** [comment]
hi, is this issue still relevant ? Looks strange but maybe this is because i do not load the buffers with `buffer` if the file already exists but only editing them, so maybe its creating new buffers all the time ?

**Comment by kyazdani42** [comment]
hi @Gelio, sorry i overlooked a lot of old issues last months, i'm getting back to them bit by bit (takes some time to manage such a project alone :smile:).
Does the tab issue still happen ? I could give a try at fixing it.

**Comment by Gelio** [comment]
Thanks for reaching back! The issue seems to be solved now, I can no longer reproduce it with the current version.

I reckon we can close this :smile: 

Thanks for all the hard work on this plugin! I've been a heavy user of it!

Side note: this week I have started using netrw to explore the directory structure and found it a bit easier to manage when dealing with deeply nested directories because I can look at one directory at a time and see all items in a full pane, instead of a small side window. I miss the icons, diagnostics, and git information that nvim-tree provides, though :slightly_frowning_face: I would love to be able to use nvim-tree in a large split-pane already focused on the parent directory of the current file, with `o` replacing the nvim-tree buffer with the current file (call it netrw-mode).

EDIT: lol :joy: I just looked at f8f297acbf656778c31a0628bbe140829ec6230a and it seems to be exactly what I am looking for! That's awesome, thank you!
